reinforcement_learning/train_dialogue_policy.py

```diff
@@ -0,0 +1,211 @@
+    parser = argparse.ArgumentParser(prog="train_dialogue_policy.py")
+    parser.add_argument(
+        "-c",
+        "--config",
+        default=DEFAUTL_RL_CONFIG,
+        help="Path to the configuration file.",
+    )
+    participants_group = parser.add_argument_group("Participants")
+    participants_group.add_argument(
+        "--usersim_config",
+        help="Path to the user simulator configuration file.",
+    )
+    participants_group.add_argument(
+        "--agent_config",
+        help="Path to the agent configuration file.",
+    )
+    model_group = parser.add_argument_group("Model")
+    model_group.add_argument(
+        "--model_type",
+        type=str,
+        help="Type of the model.",
+        dest="model.model_type",
+    )
+    model_group.add_argument(
+        "--policy_input_size",
+        type=int,
+        help="Size of the input vector.",
+        dest="model.policy_input_size",
+    )
+    model_group.add_argument(
+        "--hidden_size",
+        type=int,
+        help="Size of the hidden layer.",
+        dest="model.hidden_size",
+    )
+    model_group.add_argument(
+        "--model_path",
+        type=str,
+        help="Path to the model.",
+        dest="model.model_path",
+    )
+    run_options_group = parser.add_argument_group("Run options")
+    run_options_group.add_argument(
+        "--use_intents",
+        action="store_true",
+        help="Whether or not to include intents in the state representation.",
+    )
+    run_options_group.add_argument(
+        "--no_nlu",
+        action="store_true",
+        help="Whether or not to use NLU.",
+    )
+    run_options_group.add_argument(
+        "--turn_penalty",
+        type=float,
+        help="Penalty for each turn.",
+    )
+    dqn_hyperparameters_group = parser.add_argument_group(
+        "DQN Hyperparameters"
+    )
+    dqn_hyperparameters_group.add_argument(
+        "--batch_size",
+        type=int,
+        help="Batch size.",
+        dest="hyperparams.batch_size",
+    )
+    dqn_hyperparameters_group.add_argument(
+        "--dqn-gamma",
+        type=float,
+        help="Discount factor.",
+        dest="hyperparams.gamma",
+    )
+    dqn_hyperparameters_group.add_argument(
+        "--eps_start",
+        type=float,
+        help="Starting value of epsilon.",
+        dest="hyperparams.eps_start",
+    )
+    dqn_hyperparameters_group.add_argument(
+        "--eps_end",
+        type=float,
+        help="Ending value of epsilon.",
+        dest="hyperparams.eps_end",
+    )
+    dqn_hyperparameters_group.add_argument(
+        "--eps_decay",
+        type=float,
+        help="Epsilon decay.",
+        dest="hyperparams.eps_decay",
+    )
+    dqn_hyperparameters_group.add_argument(
+        "--num_episodes",
+        type=int,
+        help="Number of episodes.",
+        dest="hyperparams.num_episodes",
+    )
+    dqn_hyperparameters_group.add_argument(
+        "--tau",
+        type=float,
+        help="Tau.",
```

Comment:
Could you make it a little bit more descriptive?

Reply:
Done.